tools/wait_for_healthcheck.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+# Waiting for a running container to become healthy.
+# Waits if no container was found with reason "no such object".
+# Waits if the container does not healthcheck configured.
+# Fails if this script runs for too long with the reason "Killed by timeout".
+#
+# Usage example:
+# ./tools/wait_for_healthcheck.sh mongooseim-riak
+# or
+# ./tools/wait_for_healthcheck.sh "$CONTAINER"
+#
+# with 5 seconds timeout:
+# TIMEOUT=5 ./tools/wait_for_healthcheck.sh "$CONTAINER"
+
+set -e
+
+if [ "$#" -ne 1 ]; then
+    exit "Illegal number of parameters"
+fi
+CONTAINER="$1"
+
+# Default timeout is 1 minute
+TIMEOUT="${TIMEOUT:-60}"
+
+# Stop waiting after timeout using a background task
+MAIN_PID=$BASHPID
+(sleep "$TIMEOUT"; echo ""; echo "Killed by timeout"; kill $MAIN_PID) &
```

Comment:
Why go through the complications of setting up a separate timeout process? We can achieve less fidelity but much more clarity with simply manipulating the loop:

```bash
for i in $(seq ${TIMEOUT}); do
    if [ $(health_status "$CONTAINER")"" = "\"healthy\"" ]; then
        echo "\nWaiting is done"
        exit 0
    fi
    echo -n "."
    sleep 1
done
echo "\nKilled by timeout"
exit 1
```

Reply:
Or, with the same amount of fidelity:

Reply:
Or, with the same amount of fidelity:

```

Reply:
Or, with around the same amount of fidelity with a bit less clarity:

```bash
DIE_AFTER=$(( $(date '+%s') + ${TIMEOUT} ))
until [ $(health_status "$CONTAINER")"" = "\"healthy\"" ]; do
    if [ "$(date '+%s')" -gt "${DIE_AFTER}" ]; then
        echo "\nKilled by timeout"
        exit 1
    fi
    echo -n "."
    sleep 1
done
echo "\nWaiting is done"
```

Reply:
ok, I like your code more.
We can use also `timeout` utility.

Reply:
There are two main problems with `timeout`:
* still requires trapping exists, otherwise SIGINT from user will be lost
* not present on macOS unless you install coreutils (and then present as `gtimeout`

Reply:
`echo -e "\n"`
heh, it's tricky

Reply:
Well just `echo` then :)

Reply:
I definitely can.
It's also possible that `kill` wouldn't kill the process if it hanged in docker call, anyway.

Reply:
OK, rewrote using cycle.
blocking docker is super rare, I would care, if it would cause problems for _me_.
Until then it should be enough.